src/twisted/newsfragments/7922.bugfix

```diff
@@ -0,0 +1 @@
+IService now properly requires the 'runner', 'name' and 'parent' attributes.
```

Comment:
Please use the full python path for the IService reference twisted.application.service.IService

there might be multiple IService interfaces, and with the full path it is clear what was changed.

Reply:
"Properly"?  "Requires"?  Do those words mean something?

Reply:
in case it's not clear from the previous comment (which I admit is probably the most likely event), the point here is ... As a user, what am I expected to learn from these words?  Was my code broken because of the "improperness" of this interface?  Or was *my* code "improper" if it is _now_ broken as a result of this change?  For that matter, isn't it improper for Twisted to break my code without any backwards compatibility grace period?

And weren't these attributes "required" before?  I doubt you could create a usable implementation of `IService` without having them.

Also, there is no `runner` attribute: it's `running`.